ddl/index.go

```diff
@@ -131,12 +131,12 @@ func checkIndexPrefixLength(columns []*model.ColumnInfo, idxColumns []*model.Ind
 	return nil
 }
 
-func checkIndexColumn(col *model.ColumnInfo, ic *ast.IndexPartSpecification) error {
+func checkIndexColumn(col *model.ColumnInfo, columnLen int, columnName model.CIStr) error {
 	if col.Flen == 0 && (types.IsTypeChar(col.FieldType.Tp) || types.IsTypeVarchar(col.FieldType.Tp)) {
 		if col.GeneratedExprString != "" {
 			return errors.Trace(errWrongKeyColumnFunctionalIndex.GenWithStackByArgs(col.GeneratedExprString))
 		}
-		return errors.Trace(errWrongKeyColumn.GenWithStackByArgs(ic.Column.Name))
+		return errors.Trace(errWrongKeyColumn.GenWithStackByArgs(columnName))
```

Comment:
I believe we can use the name from `col *model.ColumnInfo`. Could you please check if it is possible?

Reply:
col is the table column, and columnName is from ic which is for index column, so it cannot be used from col. I have changed the parameter name to be more clear.

Reply:
There is no difference between "index column name" and "table column name". At the time calling this function, their names are equal to each other.

Reply:
@tangenta Yes, I've checked the code and find the name is the same, but about the length, it seems we cannot remove, so I will remove 1 parameter.